screens/ActivitiesScreen.tsx

```diff
@@ -1,15 +1,174 @@
-import React from "react"
+import EventsList from "@components/EventsList"
+import EventsMap, { MapRefMethods } from "@components/EventsMap"
+import { Ionicon } from "@components/common/Icons"
+import EventTabBar from "@components/tabBarComponents/EventTabBar"
+import { useTrackUserLocation } from "@hooks/UserLocation"
+import { CurrentUserEvent, EventMocks } from "@lib/events"
+import {
+  UserLocationTrackingUpdate,
+  requestLocationPermissions
+} from "@lib/location/UserLocation"
+import React, { useEffect, useRef } from "react"
+import {
+  ImageBackground,
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+  View
+} from "react-native"
+import { Icon } from "react-native-elements"
 import EventDetails from "./EventDetails/EventDetails"
 import { EventMocks } from "@lib/events"
 
-const MARKER_SIZE = 80
+const MARKER_SIZE = 60
 
 const ActivitiesScreen = () => {
+  const appRef = useRef<MapRefMethods | null>(null)
+  let doesWork = false
+  const givenUserLocation = useTrackUserLocation()
+
+  const recenterThing = () => {
+    if (givenUserLocation?.status === "success") {
+      appRef.current?.recenterToLocation({
+        latitude: givenUserLocation?.location.coordinates.latitude,
+        longitude: givenUserLocation?.location.coordinates.longitude
+      })
+    }
+  }
+  const events: CurrentUserEvent[] = [
+    EventMocks.Multiday,
+    EventMocks.NoPlacemarkInfo,
+    EventMocks.PickupBasketball
+  ]
+
+  useEffect(() => {
+    ;(async () => {
+      const status = await requestLocationPermissions()
+      if (status === false) {
+      } else {
+        doesWork = true
+      }
+    })()
+  }, [])
+
+  function checkUserLocation (
```

Comment:
This seems to really be for retrieving the initial region. In that case, maybe you should wait for the location to be resolved before even calling this thing. Otherwise, it's possible that there could be a race condition where the map initially centers on lat 30, lng 30 even if we have the user's location (better, we could have a test for this!).